core_backend/app/routers/manage_content.py

```diff
@@ -99,51 +145,121 @@ async def delete_content(
     Delete content endpoint
     """
     record = await get_content_from_db(
-        content_id,
+        content_text_id,
         asession,
     )
 
     if not record:
         raise HTTPException(
-            status_code=404, detail=f"Content id `{content_id}` not found"
+            status_code=404, detail=f"Content id `{content_text_id}` not found"
         )
-    await delete_content_from_db(content_id, asession)
+    await delete_content_from_db(content_text_id, record.content_id, asession)
 
 
-@router.get("/{content_id}", response_model=ContentRetrieve)
+@router.get(
+    "/{content_text_id}", response_model=Union[ContentRetrieve, List[ContentRetrieve]]
+)
 async def retrieve_content_by_id(
-    content_id: int,
+    content_text_id: int,
     readonly_access_user: Annotated[
         AuthenticatedUser, Depends(get_current_readonly_user)
     ],
     asession: AsyncSession = Depends(get_async_session),
-) -> ContentRetrieve:
+    language: Optional[str] = None,
+) -> Union[ContentRetrieve, List[ContentRetrieve]]:
     """
     Retrieve content by id endpoint
     """
 
-    record = await get_content_from_db(content_id, asession)
+    content = await get_content_from_db(content_text_id, asession)
 
-    if not record:
+    if not content:
+        raise HTTPException(
+            status_code=404, detail=f"Content id `{content_text_id}` not found"
+        )
+
+    if language:
+        record = await get_content_from_content_id_and_language(
+            content.content_id, int(language), asession
+        )
+        if not record:
+            raise HTTPException(
+                status_code=404,
+                detail=f"""Content `{content_text_id}`
+                with language id `{language}` not found""",
+            )
+        return _convert_record_to_schema(record)
+    else:
+        records = await get_all_languages_version_of_content(
+            content.content_id, asession
+        )
+        return [_convert_record_to_schema(record) for record in records]
+
+
+async def is_content_and_language_valid(
+    content: ContentTextCreate,
+    asession: AsyncSession,
+    is_edit: bool = False,
+    is_updated_language: bool = True,
+) -> bool:
+    contents = await get_all_languages_version_of_content(
+        content.content_id, asession=asession
+    )
+    if len(contents) < 1:
+        if is_edit or content.content_id != 0:
```

Comment:
I see why we're checking for `is_edit`: if we're trying to update a content that doesn't exist (by ID) it shouldn't work.

But what is the content_id != 0 part checking for?

Reply:
We are adding a new content text only if content.content_id==0. If not, we add the content text to an existing content_id. So, when we are editing, we need to make sure that content.content_id!=0. But when creating, we can allow content.content_id==0.